extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/WhiteListBasedConverter.java

```diff
@@ -0,0 +1,288 @@
+   */
+  private List<String> getOrderedDimValues(ServiceMetricEvent event)
+  {
+    String prefixKey = getPrefixKey(event.getMetric(), whiteListDimsMapper);
+    if (prefixKey == null) {
+      return null;
+    }
+    ImmutableList.Builder<String> outputList = new ImmutableList.Builder();
+    Set<String> dimensions = whiteListDimsMapper.get(prefixKey);
+    if (dimensions == null) {
+      return Collections.emptyList();
+    }
+    for (String dimKey : dimensions) {
+      String dimValue = (String) event.getUserDims().get(dimKey);
+      if (dimValue != null) {
+        outputList.add(GraphiteEmitter.sanitize(dimValue));
+      }
+    }
+    return outputList.build();
+  }
+
+  /**
+   * @param serviceMetricEvent druid metric event to convert
+   *
+   * @return <tt>null</tt> if the event is not white listed, otherwise return {@link GraphiteEvent}
+   * <p>
+   * The metric path of the graphite event is:
+   * <namespacePrefix>.[<druid service name>].[<druid hostname>].<white-listed dimensions>.<metric>
+   * <p/>
+   * The order of the dimension is the order returned by {@code getOrderedDimValues()}
+   * Note that this path will be sanitized by replacing all the `.` or space by `_` {@link GraphiteEmitter#sanitize(String)}
+   * </p>
+   */
+
+  @Override
+  public GraphiteEvent druidEventToGraphite(ServiceMetricEvent serviceMetricEvent)
+  {
+    if (!this.isInWhiteList(serviceMetricEvent)) {
+      return null;
+    }
+    final ImmutableList.Builder<String> metricPathBuilder = new ImmutableList.Builder<>();
+    metricPathBuilder.add(GraphiteEmitter.sanitize(this.getNamespacePrefix()));
+    if (this.isIgnoreServiceName() == false) {
+      metricPathBuilder.add(GraphiteEmitter.sanitize(serviceMetricEvent.getService()));
+    }
+    if (this.isIgnoreHostname() == false) {
+      metricPathBuilder.add(GraphiteEmitter.sanitize(serviceMetricEvent.getHost()));
+    }
+    metricPathBuilder.addAll(this.getOrderedDimValues(serviceMetricEvent));
+    metricPathBuilder.add(GraphiteEmitter.sanitize(serviceMetricEvent.getMetric()));
+
+    final GraphiteEvent graphiteEvent = new GraphiteEvent(
+        Joiner.on(".").join(metricPathBuilder.build()),
+        String.valueOf(serviceMetricEvent.getValue()),
+        TimeUnit.MILLISECONDS.toSeconds(serviceMetricEvent.getCreatedTime().getMillis())
+    );
+    return graphiteEvent;
+  }
+
+  /**
+   * @code DEFAULT_DIM_WHITE_LIST_MAPPER is a white list of metric->dimensions mappings.
+   * Key is the metric name or the metric's prefix.
+   * Value is a list of metric's dimensions names.
+   * The order of the dimension name is important, it will be used to build the graphite metric path.
+   * For instance we have dimension type is nested under dimension dataSource -> prefix.dataSource.queryType.metricName
+   */
+  static final ImmutableSortedMap<String, Set<String>> DEFAULT_DIM_WHITE_LIST_MAPPER
```

Comment:
if we switch to reading whitelist from file then this can also go to a file packaged with this extension jar by default and documentation can just link to that file.